### DISCUSSION TITLE
RenderingLayerMask error

### DISCUSSION THREAD

**Discussion by throwyourglock** [jiaozi158/UnitySSGIURP#13] [Q&A] [answered]
Unity version 2023.1.19. URP version 15.0.7.
I get the following error when importing the package: error CS0246: The type or namespace name 'RenderingLayerMask' could not be found (are you missing a using directive or an assembly reference?)
Tried to change the code somehow, these compile errors dissapeared but the renderer feature didn't work.
Thanks in advance for your answer.
![1](https://github.com/user-attachments/assets/a3bbf97d-b2f4-48e8-9e7e-c584c1bb6a3e)

**Comment by jiaozi158** [comment, marked as answer]
Hi, thanks for reporting the issue, and [it should be fixed now](https://github.com/jiaozi158/UnitySSGIURP/commit/3591b5c01df86d8ee523b1810a7096fb7af96d84)!

> The type or namespace name 'RenderingLayerMask' could not be found

Looks like the missing "RenderingLayerMask" enum was added in Unity 2023.3, not 2023.1.

Do you need to control SSGI with Rendering Layers in 2023.1? If so, I might check if it's actually possible when I have time.

**Reply by throwyourglock** [reply]
Hi! Thanks for your help. I installed the new package. There are no errors, but the GI still doesn't work: when the "Override Ambient Lighting" option was initially enabled in the RenderFeature, there was basically no lighting on the scene; after turning it off, the lighting appeared, but remained the same as it was without the GI. When "Override Ambient Lighting" is turned on again, everything on the scene simply stops (like with zero time scale). Tried to tweak all the settings of the volume asset, but that also had no effect.
I don't use Rendering layers. If this is important, then the API is DX11, Rendering path is Deferred.

**Reply by jiaozi158** [reply]
Hi, I tested SSGI again in 2023.1.19f1 and it seems to work fine.

![SSGI](https://github.com/user-attachments/assets/cab7230d-37da-4b78-87c8-8e66592eb821)

In some cases like outdoor scenes with simple geometries, SSGI won't "change" the lighting appearance too much, as the result from SSGI is similar to the global ambient probe.

You may try testing with an emissive object and that should tell you if SSGI is working properly.

> When "Override Ambient Lighting" is turned on again, everything on the scene simply stops (like with zero time scale).

That sounds like a bug, can you post a screenshot of it?

**Reply by throwyourglock** [reply]
I tried running the scene with the GI effect on a different render preset and it worked. It turned out that the problem was that the "Light Probe System" option was turned as "Probe Volumes". When it is set to "Light Probe Groups", then everything is fine. Here is the demos where I'm switching this option.
![tst1](https://github.com/user-attachments/assets/a4f70b71-02e9-46e0-8a7f-4af58975ad91)
![tst2](https://github.com/user-attachments/assets/cb111f7d-76ab-4bce-9c29-a95e762369c1)

**Reply by jiaozi158** [reply]
Thanks again for reporting this!

Maybe the Probe Volumes (APV) system in 2023.1 is different than in Unity 6, I'll look into that when I have time.

**Reply by throwyourglock** [reply]
Always happy to help solve a problem! Thanks for the quick fix and great SSGI. Good luck with your development!

**Reply by throwyourglock** [reply]
Hi! It's me again. 
I just tried to build the project for the first time with SSGI enabled and got the following errors.
![image](https://github.com/user-attachments/assets/5254d342-f123-41b6-84b1-fc68a831d1fb)
The project builds successfully when the SSGI Renderer Feature is removed.

**Reply by jiaozi158** [reply]
Thanks for reporting, the issue should be fixed now.

It looks like the APV function `AddNoiseToSamplingPosition(3 parameters)` is different in 2023.1 (2 parameters) and caused the compiler to throw an error.

Feel free to let me know if there're other issues!

**Reply by throwyourglock** [reply]
Thanks a lot for fixing. Now it compiles perfectly!
